spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/impl/ClassFactory.java

```diff
@@ -83,6 +83,7 @@ public ICodeBaseLocator createFilesystemCodeBaseLocator(String pathName) {
         // It's not fatal if we can't.
         try {
             pathName = new File(pathName).getCanonicalPath();
+            return FilesystemCodeBaseLocator.fromCanonicalPath(pathName);
```

Comment:
shouldn't the try/catch be replaced with the proper  FilesystemCodeBaseLocator constructor that does the checks? In worst case getCanonicalPath is still called twice on IO errors.

Reply:
Good point! Instead of adding a flag to say if we want to canonicalize the path I should have simply removed the first call 🫤